frontend/src/metabase/visualizations/click-actions/drills/mlv2/QuickFilterDrill.tsx

```diff
@@ -1,146 +1,112 @@
-      section: "filter",
-      buttonType: "token-filter",
+  const drillInfo = Lib.quickFilterDrillDetails(drill);
+
+  return operators.map(operator => ({
+    name: operator,
+    title: operator,
+    section: "filter",
+    sectionDirection: "row",
+    buttonType: "token-filter",
+    question: () => applyDrill(drill, operator),
+    ...getActionOverrides(question, drill, drillInfo, operator),
+  }));
+};
 
-      question: () => applyDrill(drill, operator),
+function getActionOverrides(
+  question: Question,
+  drill: Lib.DrillThru,
+  { query, column, stageIndex, value }: Lib.QuickFilterDrillThruDetails,
+  operator: Lib.QuickFilterDrillThruOperator,
+): Partial<ClickAction> {
+  if (Lib.isDate(column) && value != null) {
+    const action: Partial<ClickAction> = {
+      sectionTitle: t`Filter by this date`,
+      sectionDirection: "column",
+      buttonType: "horizontal",
+    };
 
-      extra: () => ({
-        valueType: columnValueType,
-        columnName: clicked.column?.display_name,
-      }),
+    switch (operator) {
+      case "=":
+        return { ...action, title: t`On` };
+      case "≠":
+        return { ...action, title: t`Not on` };
+      case ">":
+        return { ...action, title: t`After` };
+      case "<":
+        return { ...action, title: t`Before` };
+      default:
+        return action;
+    }
+  }
 
-      ...overrides,
+  if (Lib.isString(column) && typeof value === "string") {
+    const columnName = Lib.displayInfo(query, stageIndex, column).displayName;
+    const valueTitle = getTextValueTitle(value);
+    const action: Partial<ClickAction> = {
+      sectionTitle: t`Filter by ${columnName}`,
+      sectionDirection: "column",
+      buttonType: "horizontal",
     };
-  });
-};
 
-const getValueType = (
-  value: unknown,
-  column: DatasetColumn,
-): FilterValueType => {
-  if (value == null) {
-    return "null";
+    switch (operator) {
+      case "=":
+        return {
+          ...action,
+          title: t`Is ${valueTitle}`,
+          iconText: operator,
+        };
+      case "≠":
+        return {
+          ...action,
+          title: t`Is not ${valueTitle}`,
+          iconText: operator,
+        };
+      case "contains": {
+        return {
+          ...action,
+          title: `Contains…`,
+          popover: getFilterPopover({ question, query, column, stageIndex }),
+        };
+      }
+      case "does-not-contain": {
+        return {
+          ...action,
+          title: `Does not contain…`,
+          popover: getFilterPopover({ question, query, column, stageIndex }),
+        };
+      }
+      default: {
+        return action;
+      }
+    }
   }
 
-  if (isNumeric(column)) {
-    return "numeric";
-  }
+  return {};
+}
 
-  if (isDate(column)) {
-    return "date";
+const getTextValueTitle = (value: string): string => {
+  if (value.length === 0) {
+    return t`empty`;
   }
 
-  if (isBoolean(column)) {
-    return "boolean";
+  if (value.length > 20) {
```

Comment:
Can we bring the constant back?
It does not look like a conscious decision without it.

Reply:
Fixed